core/codenvy-hosted-platform-api-impl/src/main/java/com/codenvy/api/dao/ldap/LdapUserDao.java

```diff
@@ -345,35 +321,12 @@ private String createFilter(String attribute, String value) {
     private Attributes getUserAttributesByFilter(InitialLdapContext context, String filter) throws NamingException {
         final SearchControls controls = new SearchControls();
         controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
-        NamingEnumeration<SearchResult> enumeration = null;
-        try {
-            enumeration = context.search(containerDn, filter, controls);
+        try (CloseableSupplier<NamingEnumeration<SearchResult>> enumSup = deferClose(context.search(containerDn, filter, controls))) {
```

Comment:
I don't like deferClose method name. As it is verb it says that it does something, but it does not. It just creates wrapper. So method says that context will be closed, but actually without try with resources or similar construction context won't be closed.